example-osgi/pom.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.algolia</groupId>
+    <artifactId>example-osgi</artifactId>
+    <version>3.9.0</version>
+    <packaging>bundle</packaging>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>4.2.1</version>
+                <extensions>true</extensions>
+                <executions>
+                    <execution>
+                        <id>bundle-manifest</id>
+                        <phase>process-classes</phase>
+                        <goals>
+                            <goal>manifest</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>com.algolia.example-osgi</Bundle-SymbolicName>
+                        <Bundle-Name>example-osgi</Bundle-Name>
+                        <Bundle-Activator>com.algolia.example.osgi.DemoBundle</Bundle-Activator>
+                        <Export-Package>com.algolia.example.osgi</Export-Package>
+                        <Import-Package>com.algolia.search,org.osgi.framework,*</Import-Package>
+                    </instructions>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.osgi.core</artifactId>
+            <version>1.4.0</version>
+        </dependency>
+        <dependency>
+            <groupId>com.algolia</groupId>
+            <artifactId>algoliasearch-apache-uber</artifactId>
+            <version>3.9.0</version>
+        </dependency>
+    </dependencies>
+</project>
```

Comment:
nit: missing blank line

Reply:
> Thanks for all the research on this topic! Looking great. 💯
> 
> Question: Do we plan to deploy the bundle along the other jars or do we let our users build it by their-self?

Thanks for the kind words. :)

No I don't think we should deploy it to Maven Central for now since this is just a demo OSGi application to ensure our Apache Uber JAR is a valid OSGi bundle. In other words, no one will ever need to import this `example-osgi` as a bundle. It's just, well, an example code one can use to test OSGi.

For now, this PR is just a first step: showing how to build and integrate our unmodified Apache Uber JAR into an OSGi application. In the future, if we have the time to investigate if OSGi applications have a common way to fetch dependencies from Maven repositories, we'll do that as a second step.